### DISCUSSION TITLE
CSV file is detected as FILE

### DISCUSSION THREAD

**Discussion by botzill** [directus/directus#5578] [Archive]
Hi.

I have the latest version `.64` and I see that `.csv` file is detected as `FILE`:

<img width="921" alt="Screen Shot 2021-05-08 at 09 05 16" src="https://user-images.githubusercontent.com/4686040/117528788-b3415000-afdc-11eb-9144-7f9594aed0b9.png">

While the other CSV files I upload are detected as CSV:

<img width="628" alt="Screen Shot 2021-05-08 at 09 06 20" src="https://user-images.githubusercontent.com/4686040/117528803-c3592f80-afdc-11eb-9e15-c5a736f0b9b2.png">

<img width="962" alt="Screen Shot 2021-05-08 at 09 07 48" src="https://user-images.githubusercontent.com/4686040/117528830-dff56780-afdc-11eb-9905-b9497a3c053e.png">

I see that the one that is detected as `FILE` has this `.bin` extension in file name:

<img width="929" alt="Screen Shot 2021-05-08 at 09 08 12" src="https://user-images.githubusercontent.com/4686040/117528851-fdc2cc80-afdc-11eb-9cc5-7007bd30e7d0.png">

But eventually, I can download that and process it as CSV.

Any idea why?

**Comment by rijkvanzanten** [comment]
We're extracting the mime type from the uploaded file stream, so there must be a difference in how both files are encoded! 🤔 

https://github.com/directus/directus/blob/main/api/src/controllers/files.ts#L47
